content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+- `--experimental` and `--early-access-feature` used in `migrate`
+- `--force`/`-f` used in `db push`
+- `--experimental-reintrospection` and `--clean` used in `db pull`
+
+The outdated use of `db push --force` can be replaced with the newer implementation `db push --accept-data-loss`.
+
+All other flags are from previous APIs and are no longer needed.
+
+### Removal of the <inlinecode>beforeExit</inlinecode> hook from the library engine
+
+The `beforeExit` hook has been removed from the Prisma library engine. Instead of this hook we recommend using built-in Node.js exit events.
+
+The following code with Prisma 4:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+prisma.$on('beforeExit', exitHandler)
+```
+
+Could become:
+
+```js
+const exitHandler = () => {
+  // your exit handler code
+}
+
+process.on('exit', exitHandler)
+process.on('beforeExit', exitHandler)
+process.on('SIGINT', exitHandler)
+process.on('SIGTERM', exitHandler)
+process.on('SIGUSR2', exitHandler)
+```
+
+### Removal of deprecated <inlinecode>prisma2</inlinecode> executable
+
+When we released Prisma 2, the `prisma2` executable was used in order to differentiate from Prisma 1. In a later release, the `prisma2` cli took over the `prisma` executable name.
+
+Needless to say, the `prisma2` executable has been deprecated for some time and is now removed. If your scripts use Prisma CLI as `prisma2`, please replace it with simply `prisma`.
+
+### Removal of deprecated <inlinecode>experimentalFeatures</inlinecode> property
+
+The `previewFeatures` field of the [generator block](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#fields-1) used to be called `experimentalFeatures`. We are finally removing that deprecated property.
+
+In Prisma 5, you will need to update references of `experimentalFeatures` to `previewFeatures` manually or use the new code action in the Prisma VSCode extension.
+
+### <inlinecode>migration-engine</inlinecode> renamed to <inlinecode>schema-engine</inlinecode>
+
+The engine responsible for commands like `prisma migrate` and `prisma db` has been renamed from `migration-engine` to `schema-engine` to better describe its use. For many users, no changes will be required. However, if you need to explicitly include or exclude this engine file for any reason, you will need to update your code references.
+
+#### Example with the Serverless Framework
+
+One example we have seen is projects using the Serverless Framework. In these instances, you will need to update any patterns that reference `migration-engine` to instead reference `schema-engine`.
+
+```yaml highlight=6;delete|7;add
+package:
+  patterns:
+    - '!node_modules/.prisma/client/libquery_engine-*'
+    - 'node_modules/.prisma/client/libquery_engine-rhel-*'
+    - '!node_modules/prisma/libquery_engine-*'
+    - '!node_modules/prisma/migration-engine-*'
+    - '!node_modules/prisma/schema-engine-*'
```

Comment:
Not actionable in this PR, but the syntax highlighting hides the `-` here (probably considers it as `-` as in diff) 😕

Reply:
We could not use `highlight=6;delete|7;add` here to avoid this.

Reply:
Yeah true. Given accessibility problems in addition to the bug, I'm not a fan of this.

Reply:
And "accessibility" doesn't just mean screen readers (that might actually even be okay, I haven't checked the aria attributes), it's also colorblind people etc.

Here's how it looks like to users with deuteranopia:
<img width="697" alt="image" src="https://github.com/prisma/docs/assets/4923335/472147f7-28c3-48ab-a5b1-75ea1ba02f0c">

Reply:
That is a separate thing to highlight in #docs in Slack (or via an issue), we use this all over the place.

Reply:
Back to the original problem:
Is there a way to render the `-` even when using the highlight @jharrell @ruheni @nikolasburk @nilubava ?

Reply:
@aqrln how does github's diffing look with that deuteranopia filter? We should definitely improve that experience and I'm wondering if github already has and we can learn from that.

Reply:
https://github.com/prisma/docs/issues/5036 + https://github.com/prisma/docs/issues/5037

Reply:
@jharrell similarly, but it has pretty visible and contrast `+` and `-` signs to the left of the corresponding lines in the diff, color is secondary.

Reply:
We applied a "workaround" of having `--` so it now shows a `-` in the rendered view.